### PR TITLE
Update CodeClimate configuration

### PR DESCRIPTION
**Pull request by eckdanny-osi**
Updates `.codeclimate.yml` to version 2 format. Removes deprecated elements. Processes JSX files. Relaxes some of the de-dupe heuristics for JS[X]. Supports objectSpread operator in JS.

### CODE REVIEW

File: .codeclimate.yml [hunk 1]
```diff
@@ -29,19 +81,18 @@ engines:
     enabled: true
     issue_override:
       severity: major
-ratings:
```
Comment (eckdanny-osi) [inline .codeclimate.yml]: `ratings` are deprecated ([link](https://docs.codeclimate.com/v1.0/docs/advanced-configuration#section-analysis-configuration-versions))

File: .codeclimate.yml [hunk 2]
```diff
@@ -1,23 +1,75 @@
...
+  method-count:
+    enabled: true
+    config:
+      threshold: 20
+  method-lines:
+    enabled: true
+    config:
+      languages:
+        javascript:
+          threshold: 50
```
Comment (eckdanny-osi) [inline .codeclimate.yml]: increased from default 25

File: .codeclimate.yml [hunk 3]
```diff
@@ -1,23 +1,75 @@
...
+  complex-logic:
+    enabled: true
+    config:
+      threshold: 4
+  file-lines:
+    enabled: true
+    config:
+      languages:
+        javascript:
+          threshold: 350
```
Comment (eckdanny-osi) [inline .codeclimate.yml]: increased from default 250

File: .codeclimate.yml [hunk 4]
```diff
@@ -1,23 +1,75 @@
...
+        javascript:
+          threshold: 50
+  nested-control-flow:
+    enabled: true
+    config:
+      threshold: 4
+  return-statements:
+    enabled: true
+  similar-code:
+    enabled: false
```
Comment (edgibbs) [inline .codeclimate.yml]: Flip to true so Ruby gets checked

File: .codeclimate.yml [hunk 5]
```diff
@@ -1,23 +1,75 @@
...
+      - "app/javascript/**/*.test.js"
+      # - "app/javascript/**/*.jsx"
+      # - "app/javascript/**/*.js"
       - "spec"
     config:
       languages:
-      - ruby
-      - javascript
+        - ruby
+        - javascript
```
Comment (edgibbs) [inline .codeclimate.yml]: Remove javascript so we don't get a bunch of false positives

File: .codeclimate.yml [hunk 6]
```diff
@@ -1,23 +1,75 @@
...
     enabled: true
-  csslint:
+  scss-lint:
     enabled: true
   duplication:
+    enabled: true
     exclude_patterns:
-      - "app/javascript/clientIdPage/ClientIdPage.test.js"
+      - "app/javascript/**/*.test.js"
+      # - "app/javascript/**/*.jsx"
```
Comment (edgibbs) [inline .codeclimate.yml]: Just remove exclude_patterns

**Pull request merged**